src/Zadatak2/Pogadjanje.java

```diff
@@ -0,0 +1,84 @@
+/*
+  Napisati program za pogađanje slučajno generisanog broja u opsegu od 1 do 50.
+  Program treba da radi na sledeći način: Korisnik preko standardnog ulaza unosi jedan ceo broj.
+  Broj koji treba pogoditi generisati “random” u opsegu od 1 do 50.
+  Vrteti se u petlji dok ne pogodimo broj, ili budemo barem blizu (ako je razlika između brojeva 5 ili manja).
+  Taj deo sa proverom pogotka izdvojiti u zasebnu metodu, kojoj se kao argument prosleđuje broj koji tipujemo,
+  a povratna vrednost metode je 0 – ako smo pogodili, 1 – ako smo bili blizu, 2 – ako smo promašili.
+  U zavisnosti od rezultata pogađanja ispisati i odgovarajuću poruku.
+  Nakon odigrane partije proveriti da li korisnik zeli da prekine ili da nastavi igru.
+ */
+
+package Zadatak2;
+
+import java.util.InputMismatchException;
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * AUTOR: Vladimir Krekic
+ */
+
+class Pogadjanje {
+
+    private static Scanner scanner = new Scanner(System.in);
+    private static Random randomNumber = new Random();
+    private static int broj;
+
+    public static void main(String[] args) {
+
+        while(true) {
+
+            //Generise random broj izmedju 1 i 50
+            int pogodi = randomNumber.nextInt(50)+1;
+
+            System.out.print("Upisi broj izmedju 1 i 50: ");
+            //Ponavlja unos sve dok ne bude unet ceo broj izmedju 1 i 50
+            try {
+                broj = scanner.nextInt();
+                if(broj > 50 || broj < 1){
+                    System.out.println("Niste uneli broj izmedju 1 i 50");
+                    continue;
+                }
+            }catch (InputMismatchException e){
+                System.out.println("Niste uneli broj");
+                scanner.nextLine();
+                continue;
+            }
```

Comment:
Ovde treba dodati finaly i tu moze da se zatvori scanner varijabla npr. finally {
    if(scanner!=null)
        scanner.close();
}
Ili zatvoriti na drugom mestu, svakako treba to uraditi